apps/ledger-live-mobile/src/hooks/useErrorRenderData.ts

```diff
@@ -0,0 +1,132 @@
+  iconColor: string;
+  hasExportLogs: boolean;
+  onPrimaryPressOverride: (() => void) | undefined;
+};
+
+export type LedgerError =
+  | LedgerErrorConstructor<{ [key: string]: unknown }>
+  | LedgerErrorConstructor<{ currencyName: string }>;
+
+/**
+ * Nb If you are adding an override to a specific error class, please update
+ * the array below too. It's used by the DebugErrors screen to indicate errors
+ * that currently have overriden values such as a different icon, color, cta, etc.
+ */
+export const errorsWithOverrides: LedgerError[] = [
+  UserRefusedFirmwareUpdate,
+  ManagerDeviceLockedError,
+  LockedDeviceError,
+  UserRefusedDeviceNameChange,
+  UpdateYourApp,
+  BluetoothNotSupportedError,
+  DeviceNotOnboarded,
+  ImageCommitRefusedOnDevice,
+  ImageLoadRefusedOnDevice,
+];
+
+/**
+ * The rendering of the error will follow a specific layout.
+ * The wording for the title, description and CTA button are defined on the
+ * language files under the errors.ErrorClass path.
+ */
+const useErrorRenderData: (error: Error) => ErrorRenderData = error => {
+  const navigation: StackNavigationProp<ParamListBase> = useNavigation();
+
+  // Declare a default configuration that applies to any error we don't override.
+  const data: ErrorRenderData = {
+    Icon: Icons.CloseMedium,
+    iconColor: "error.c60",
+    hasExportLogs: true,
+    onPrimaryPressOverride: undefined,
+  };
+
+  // Specify the configuration for an error.
+  switch (true) {
+    case error instanceof UpdateYourApp:
+      data.onPrimaryPressOverride = () => {
+        navigation.navigate(NavigatorName.Manager, {
+          screen: ScreenName.Manager,
+          params: {
+            tab: MANAGER_TABS.INSTALLED_APPS,
+            updateModalOpened: true,
+          },
+        });
+      };
+      break;
+    case error instanceof ImageLoadRefusedOnDevice:
+    case error instanceof ImageCommitRefusedOnDevice:
+      data.Icon = Icons.CircledAlertMedium;
+      data.iconColor = "warning.c100";
+      break;
+
+    case error instanceof LockedDeviceError:
+      data.Icon = Icons.LockAltMedium;
+      data.iconColor = "neutral.c100";
+      data.hasExportLogs = false;
+      break;
+
+    case error instanceof BluetoothNotSupportedError:
+      data.Icon = Icons.UsbMedium;
+      data.iconColor = "neutral.c100";
+      data.hasExportLogs = false;
+      break;
+
+    case error instanceof UserRefusedDeviceNameChange:
+      data.Icon = Icons.WarningSolidMedium;
+      data.iconColor = "warning.c80";
+      break;
+
+    case error instanceof DeviceNotOnboarded:
+      data.Icon = Icons.InfoAltFillMedium;
+      data.hasExportLogs = false;
+      break;
+
+    case error instanceof UserRefusedFirmwareUpdate:
+      data.Icon = Icons.WarningMedium;
+      break;
+
+    case error instanceof ManagerDeviceLockedError:
+      data.Icon = Icons.LockAltMedium;
+      data.iconColor = "warning.c80";
+      data.hasExportLogs = false;
+      break;
+
+    default:
+      // Yaiza
```

Comment:
😂

Reply:
TS wanted something there, again, this PR is not for review really, it should've been a draft :trollface: